tests/integration/valkey-cli.tcl

```diff
@@ -833,6 +843,37 @@ if {!$::tls} { ;# fake_redis_node doesn't support TLS
         close_cli $fd
     }
 
+    test "valkey-cli make sure selected db survives connection drops" {    
+        set fd [open_cli]
+        
+        # Target database
+        set db 9
+
+        # Select the database        
+        assert_equal "OK" [run_command $fd "select $db"]
+
+        # Kill all normal clients, to disconnect vlakey-cli client
+        r client kill type normal
+        after 10
+
+        # Trigger reconnect
+        write_cli $fd "ping"
+
+        assert_equal "OK" [run_command $fd "set x 1"]        
+        assert_equal "OK" [r select $db]        
+        assert_equal "1" [r get x]
+        
+        set client_info [run_command $fd "client info"]
+
+        # set actual_db [get_client_info_field $client_info "db"]
+
+        # # Validate that the selected DB matches the reported DB
+        # assert_equal $db $actual_db
+        
+        # r flushall
+        # r select 0
```

Comment:
Unused stuff. Delete it?

Btw, the reason `get_client_info_field` doesn't already exist is that other test cases just do it this way:

```tcl
regexp {db=(\d+)} $client_info _ actual_db
```